### PR TITLE
feat(activate): add --fetchVariable flag to fetch job variables server-side

### PR DESCRIPTION
**Pull request by georgios-goulos**
## Summary

- Adds a `--fetchVariable` flag to `c8ctl activate jobs` that takes a comma-separated list of variable names (e.g. `--fetchVariable scopeInput,projectName`)
- The names are sent in the Activate Jobs request body as the API's `fetchVariable` field, so the server returns **only** the requested variables for each activated job instead of the full payload
- The requested variables are also surfaced in the command output
- Rejects a provided-but-empty `--fetchVariable` (see [Empty-value handling](#empty-value-handling) below) so users can't unknowingly fetch the full variable set

## Context

Required by `governance-phase-activation` in `process-os` as part of the variable→file bridge described in the scope capture spec (in https://github.com/camunda/process-os/pull/278 see `2026-07-07-scope-capture-and-discovery-config-rework.md`, §4.1.1). The bridge needs to read the `scopeInput` variable from activated `camunda-init` jobs without raw REST calls; `c8ctl` handles auth transparently.

Fetching every variable on job activation is a real cost in agent workflows — full-value variables (e.g. `scopeInput` blobs) are fetched and paid for even when only one is needed. The API's `fetchVariable` field is the correct lever to filter **server-side**, which is what this flag exposes.

## Empty-value handling

Per the [Activate Jobs REST spec](https://docs.camunda.io/docs/apis-tools/orchestration-cluster-api-rest/specifications/activate-jobs/), an *empty* `fetchVariable` list means "return all visible variables". Silently sending an empty list would defeat the purpose of the flag: the user would think they were reducing the payload while still fetching everything.

So when `--fetchVariable` is provided but resolves to no usable names after splitting/trimming (e.g. `--fetchVariable " , , "`), the command now fails with a clear error and a non-zero exit, consistent with the other flag guards in this handler and the `--fetchVariables` guard in `process-instances.ts`.

Note: a literal empty string (`--fetchVariable ""`) is normalized to `undefined` by the framework's `deserializeFlags` before the handler runs, so it stays indistinguishable from omitting the flag (no filter applied). Only non-empty values that collapse to an empty list are rejected. Not passing the flag at all remains the legitimate "no filtering" path.

## Naming

The flag is `--fetchVariable` (singular), matching the Activate Jobs API spec exactly. It is intentionally distinct from the existing `--fetchVariables` (plural) on `create process-instance`, which maps to a different API.

## Test plan

- [x] `npm run build` — clean (0 lint/type/tsc errors)
- [x] `npm run test:unit` — new/updated behavioural tests pass:
  - dry-run omits `fetchVariable` by default
  - dry-run sends trimmed name array; drops empty tokens and keeps the rest
  - provided-but-empty value (`" , , "`) fails with a non-zero exit and actionable error
  - empty string is treated as unset (framework normalizes it away)
  - filter reaches the wire (asserted against a mock server); variables rendered in JSON and text modes

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/commands/jobs.ts [hunk 1]
```diff
@@ -88,6 +88,12 @@ export const activateJobsCommand = defineCommand(
 			: 10;
 		const timeout = flags.timeout ? parseInt(flags.timeout, 10) : 60000;
 		const worker = flags.worker || "c8ctl";
+		// `--variables` is registered as a string type in parseArgs (for
+		// `complete`/`fail`/`publish job --variables '{...}'`, which
+		// `deriveParseArgsOptions` promotes globally) but is used here as a
+		// boolean display toggle. Read argv directly so detection is
+		// position-independent (mirrors `get pi --variables`).
+		const includeVariables = process.argv.includes("--variables");
```
[the remark this answers is not in the document]
Comment (georgios-goulos) [reply src/commands/jobs.ts]: `const includeVariables = process.argv.includes("--variables");` appears in [process-instances.ts](https://github.com/camunda/c8ctl/blob/4c4e1ac93553f7228d4848f8f95729bddad1c1e5/src/commands/process-instances.ts). If we change it here, we should change it consistently everywhere and that should be in a separate PR. I will leave this up to reviewers to assess.

**Review by tmetzke — COMMENTED** [review]
❓ Can we add support for restricting the variables to fetch by name as the API supports? Fetching all available variables can become quite a token drain when used with agents, especially since we fetch full-value variables on job activation.

**Comment by georgios-goulos** [comment]
> ❓ Can we add support for restricting the variables to fetch by name as the API supports? Fetching all available variables can become quite a token drain when used with agents, especially since we fetch full-value variables on job activation.

@tmetzke I am happy to do that. Actually, I would prefer repurposing this PR so it only adds support for fetching variables by name. I will drop the current implementation and I will add support for `fetchVariable: string[]`.

**Comment by tmetzke** [comment]
That makes sense to me 👍 Fetching all variables on activation is not a good idea in most cases anyway 🤷

File: src/commands/jobs.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (georgios-goulos) [reply src/commands/jobs.ts]: No longer relevant

File: tests/unit/jobs-behaviour.test.ts
Comment (tmetzke) [inline tests/unit/jobs-behaviour.test.ts:1]: ❓ Should we add a test for empty `fetchVariable` to lock in behavior?

Comment (georgios-goulos) [reply tests/unit/jobs-behaviour.test.ts:1]: There are some other test gaps I am addressing now, I will make sure this is captured too

Comment (tmetzke) [reply tests/unit/jobs-behaviour.test.ts:1]: OK, sorry, I just now saw the draft state 😅

Comment (georgios-goulos) [reply tests/unit/jobs-behaviour.test.ts:1]: No worries! It's always good to surface gaps, I could have missed it and created another unnecessary review cycle 🙂

File: src/commands/jobs.ts [hunk 2]
```diff
@@ -88,6 +88,14 @@ export const activateJobsCommand = defineCommand(
...
 		const timeout = flags.timeout ? parseInt(flags.timeout, 10) : 60000;
 		const worker = flags.worker || "c8ctl";
+		const fetchVariableNames = flags.fetchVariable
+			?.split(",")
+			.map((name) => name.trim())
+			.filter(Boolean);
+		const fetchVariable =
+			fetchVariableNames && fetchVariableNames.length > 0
+				? fetchVariableNames
+				: undefined;
```
[the remark this answers is not in the document]
Comment (georgios-goulos) [reply src/commands/jobs.ts]: agreed, and fixed. `--fetchVariable` now rejects a provided-but-empty value: when it resolves to no usable names after splitting/trimming, the command fails with 

```
--fetchVariable was given no valid variable names
```

and a non-zero exit, rather than silently omitting the field and fetching the full variable set. This matches the existing guards in this handler (--maxJobsToActivate, --timeout) and the --fetchVariables guard in process-instances.ts.

One caveat worth noting: a literal empty string (`--fetchVariable ""`) is normalized to `undefined` by the framework's `deserializeFlags` before the handler runs, so it's indistinguishable from omitting the flag and can't be caught at this layer without introducing a `process.argv` workaround or a cross-cutting framework change. The guard therefore rejects any non-empty value that collapses to an empty list (e.g. " , , ", ",,"), which is the realistic footgun; a bare "" behaves exactly like not passing the flag. If there is a requirement for supporting this in the future, it can be tackled in a separate PR.

File: tests/unit/jobs-behaviour.test.ts [hunk 3]
```diff
@@ -198,6 +198,75 @@ describe("CLI behavioural: activate jobs", () => {
...
+			"my-job-type",
+			"--fetchVariable",
+			" , , ",
+		);
+
+		assert.strictEqual(result.status, 0, `stderr: ${result.stderr}`);
+		const body = asRecord(parseJson(result).body, "dry-run body");
+		// Every token trims to empty, so nothing survives filtering.
+		assert.strictEqual(body.fetchVariable, undefined);
+	});
```
[the remark this answers is not in the document]
Comment (georgios-goulos) [reply tests/unit/jobs-behaviour.test.ts]: Updated. The " , , " test now asserts a non-zero exit code and matches the error message. The empty-string case can't assert an error. The framework normalizes "" to `undefined` before the handler, so it's treated as unset. That test now documents that behavior explicitly (exit 0, fetchVariable absent from the body) rather than asserting an error that we can't raise here.

**Review by tmetzke — APPROVED** [review]
🚀 Awesome
